src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -115,16 +116,20 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, cluster_type,
             storage_account_key = key
 
     # Attempt to provide a default container for WASB storage accounts
-    if not storage_default_container and storage_account_endpoint \
-       and _is_wasb_endpoint(storage_account_endpoint):
+    if not storage_default_container and is_wasb:
         storage_default_container = cluster_name
         logger.warning('Default WASB container not specified, using "%s".', storage_default_container)
+    elif not storage_default_filesystem and not is_wasb:
+        storage_default_filesystem = cluster_name
+        logger.warning('Default ADLS file system not specified, using "%s".', storage_default_filesystem)
 
     # Validate storage info parameters
-    if not _all_or_none(storage_account, storage_account_key,
-                        (storage_default_container or storage_default_filesystem)):
+    if is_wasb and not _all_or_none(storage_account, storage_account_key, storage_default_container):
         raise CLIError('If storage details are specified, the storage account, storage account key, '
-                       'and either the default container or default filesystem must be specified.')
+                       'and the default container must be specified.')
+    elif not is_wasb and not _all_or_none(storage_account, storage_default_filesystem):
+        raise CLIError('If storage details are specified, the storage account, '
+                       'and the default filesystem must be specified.')
```

Comment:
please double check whether or not the error text is right. Not so sure what `_all_or_none` means really.

Reply:
Good catch! This logic is not quite accurate but cannot really be reached. I will make a little change to check none only.

Reply:
After a third thought, this is correct logic. Let's keep it as what it is now.